src/networking.c

```diff
@@ -209,6 +210,94 @@ cc_oci_tap_create(const gchar *const tap) {
 }
 
 /*!
+ * Set MTU for network interface.
+ *
+ * \param[in] ifname Interface name.
+ * \param mtu MTU to be set for the interface.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+set_interface_mtu(char *ifname, unsigned int mtu)
+{
+	int socket_fd = -1;
+	gboolean ret = false;
+	struct ifreq ifr = { 0 };
+
+	if (! ifname) {
+		return false;
+	}
+
+	socket_fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (socket_fd < 0) {
+		g_critical("Error opening socket: %s", strerror(errno));
+		goto out;
+	}
+
+	ifr.ifr_addr.sa_family = AF_INET;
+	strncpy(ifr.ifr_name, ifname, IFNAMSIZ-1);
+	ifr.ifr_mtu = (int)mtu;
+
+	if (ioctl(socket_fd, SIOCSIFMTU, (caddr_t)&ifr) < 0) {
+		g_critical("ioctl SIOCSIFMTU (set mtu) error :%s", strerror(errno));
+		goto out;
+	}
+
+	g_debug("MTU set for interface %s as %d\n", ifname, ifr.ifr_mtu);
+	ret = true;
+
+out:
+	if (socket_fd > 0) {
+		close (socket_fd);
+	}
+	return ret;
+}
+
+/*!
+ * Get MTU of the network interface.
+ *
+ * \param[in] ifname Interface name.
+ * \param[out] mtu MTU returived for the interface.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+cc_oci_get_interface_mtu(char *ifname, unsigned int *mtu)
```

Comment:
This function is almost dentical to `set_interface_mtu()`. To avoid code duplication, and ease the maintenance burden, I suggest you do something like this:

```
gboolean
handle_interface_mtu(char *ifname, unsigned int mtu, gboolean set);

gboolean
set_interface_mtu(char *ifname, unsigned int mtu, gboolean set)
{
    return handle_interface_mtu (ifname, mtu, true);
}

gboolean
get_interface_mtu(char *ifname, unsigned int mtu, gboolean set)
{
    return handle_interface_mtu (ifname, mtu, false);
}
```